### ISSUE TITLE
"At the end of the session, at least once validator was found to be offline" with server online and not in stuck

### ISSUE THREAD

**Issue by dbpatty** [paritytech/polkadot#789]
hi, 
this morning i have found my commission to 0 but i don't have done any change about that, i have tried to find my last transaction filtering in Polkascan transactions section, after commission to 0 i have setted old commission value and added some info in identity, all seems to be right. At the end of the era we are not selected for next session and 0 nominations like we have slashed , but in statistics there aren't slashes and my balance is the same.
In ui i see like i have last block  747,782 but 0 points

**Comment by dbpatty** [comment]
To try to revalidate i have rechanged commision values to 0, now my node is in waiting list with 0 nominations only self-stake, no slash in chart so i continue to understand what it was happened.

**Comment by dbpatty** [comment]
hi, i have found this info in polkascan:
https://polkascan.io/pre/kusama-cc3/event/746264-0 
So i have understood why we wasn't selected for next session but  I have a system alert that send to me messagges about warn, error on log and stuck  and other thing using RPC... but at this session my server was online and cpu and RAM OK
I would propose to wait 2 sessions without heartbeat before to go out from validatorset

**Comment by thiolliere** [comment]
> this morning i have found my commission to 0
> In ui i see like i have last block 747,782 but 0 points

Is this during the last session of an era?
Because the last session of an era has some issue in regard to commission and points. the new validator set for next era (and thus next session) is considered active in staking and rewards are wrong.
See https://github.com/paritytech/substrate/issues/4121
The incoming fix is https://github.com/paritytech/substrate/pull/4474

Though I can't tell about offline stuff, maybe the error here is not related.

**Comment by dbpatty** [comment]
many thanks for reply, now i know that what i have seen in ui it is for this reason but i don't understood why my node seemed offline so i'll not can trace this situation with alert .
Is there a way to understand or to change to go out parameters? At this moment I have 2 scripts to check node status: 1) one reads warn and err  on log  2) one uses rpc call to check some parameters

**Comment by dbpatty** [comment]
at the end, in my case it seems that if a node doesn't validate blocks or it hasn't heartbeat in last session it appears like all era  though server is perfectly online

**Comment by Mitchell-Grimes** [comment]
Do you have an way to check logs for your authored blocks? If so, it might be something similar to what I've seen with our validator. For example my validator shows on session `1,367` that we authored "0" blocks. I check the logs for the validator for the time frame. In this case it's
```
Session #1,367
Start at block 764915 
Timestamp	Jan 24, 2020, 11:28:24 AM EST
End at block 765490 
Timestamp	Jan 24, 2020, 12:27:54 PM
```

Sure enough, for the given 1 hour within the session we did not author a block (even though our machine is online, no panics, no errors, syncing the chain normally). Within that epoch(session) `1,367` we the event "SomeOffline" happened via Polkastats. Which leads me to believe 2 things:
1. With the session being only an hour and having the validator pool set to `160` and continue to grow, is it more likely to not author a block within a given session due to the randmoness of authoring blocks?
2. If statement 1 above is valid and it's the case that we will be deemed offline more frequently, does that mean the first offence registers your validator to be removed in the pool at the start of the next era?

Question `2` is because as I continued to observe the logs of sessions `1,368, 1,369,1,370` I did author blocks within logs that match the graph I showcase in the screenshot below. This means I was online for 3 hours authoring some blocks. But again, in session `1,372` I failed to author any blocks for that timeframe:
```
Session #1371
Start at Block #767227
Timestamp	Jan 24, 2020, 3:28:00 PM
End at Block #767807
Timestamp	Jan 24, 2020, 4:27:54 PM
```

This means, I indeed saw that for the second time I was unable to author a single block within the small window of the epoch.

TL:DR I checked logs because I saw a similar behavior and went to investigate. Found that in the first `epoch` (1hr period) where I didn't author any blocks I was deemed offline. Is this normal behavior to be flagged to be removed out of the pool at the start of the next ERA when this happens? Since for 6 consecutive hours I was still participating in the pool for authoring blocks before being removed from the pool.
![First_session_0blocks_authored](https://user-images.githubusercontent.com/52660893/73124897-a414cb00-3f6e-11ea-8ff9-aec8404ec80d.png)

**Comment by chrisli30** [comment]
We regularly see this imOnline:someOffline message in the blocks of our Substrate network. The validator nodes are actually running fine with stable network access. I'm following up here to see if anybody can explain it.

**Comment by rphmeier** [comment]
outdated
